core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -977,6 +985,156 @@ class CivilizationInfo {
+            GameSpeed.Marathon -> 30
+        }
+        val turnsToIncrement = when (gameInfo.gameParameters.gameSpeed) {
+            GameSpeed.Quick -> 5f
+            GameSpeed.Standard -> 6.5f
+            GameSpeed.Epic -> 14f
+            GameSpeed.Marathon -> 32f
+        }
+        gold += 5 * (gameInfo.turns / turnsToIncrement).toInt()
+
+        return gold
+    }
+
+    fun demandGold(cityState: CivilizationInfo) {
+        if (!cityState.isCityState()) throw Exception("You can only demand gold from City-States!")
+        val goldAmount = goldGainedByTribute()
+        addGold(goldAmount)
+        cityState.getDiplomacyManager(this).influence -= 15
+        cityState.addFlag(CivFlags.RecentlyBullied.name, 20)
+        cityState.updateAllyCivForCityState()
+        updateStatsForNextTurn()
+    }
+
+    fun demandWorker(cityState: CivilizationInfo) {
+        if (!cityState.isCityState()) throw Exception("You can only demand workers from City-States!")
+
+        val buildableWorkerLikeUnits = gameInfo.ruleSet.units.filter {
+            it.value.uniqueObjects.any { it.placeholderText == Constants.canBuildImprovements }
+                    && it.value.isBuildable(this)
+                    && it.value.isCivilian()
+        }
+        if (buildableWorkerLikeUnits.isEmpty()) return  // Bad luck?
+        placeUnitNearTile(cityState.getCapital().location, buildableWorkerLikeUnits.keys.random())
+
+        cityState.getDiplomacyManager(this).influence -= 50
+        cityState.addFlag(CivFlags.RecentlyBullied.name, 20)
+        cityState.updateAllyCivForCityState()
+    }
+
+    fun canGiveStat(statType: Stat): Boolean {
+        if (!isCityState())
+            return false
+        val eraInfo = getEraObject()
+        val bonuses = if (eraInfo == null) null
+            else eraInfo.allyBonus[cityStateType.name]
+        if (bonuses != null) {
+            // Defined city states in json
+            bonuses.addAll(eraInfo!!.friendBonus[cityStateType.name]!!)
+            for (bonus in bonuses) {
+                if (statType == Stat.Happiness && bonus.getPlaceholderText() == "Provides [] Happiness")
+                    return true
+                if (bonus.getPlaceholderText() == "Provides [] [] per turn" && bonus.getPlaceholderParameters()[1] == statType.name)
+                    return true
+                if (bonus.getPlaceholderText() == "Provides [] [] []" && bonus.getPlaceholderParameters()[1] == statType.name)
+                    return true
+            }
+
+        } else {
+            // compatibility mode
+            return when {
+                cityStateType == CityStateType.Mercantile && statType == Stat.Happiness -> true
+                cityStateType == CityStateType.Cultured && statType == Stat.Culture -> true
+                cityStateType == CityStateType.Maritime && statType == Stat.Food -> true
+                else -> false
+            }
+        }
+
+        return false
+    }
+
```

Comment:
After considering this for a while, I think these ~200 lines should go in a separate class. CivInfo is quite packed as it is.